### PR TITLE
Normalize rootPath for Windows (forward slashes)

### PR DESCRIPTION
**Pull request by highlevellogic**
Normalize rootPath for Windows (forward slashes)

This change should have no effect on Linux.

issue: Path must start with hostname #1491
one change in file lib/resource-mapper.js

"path" variable uses forward slashes. "rootPath" does not. IMO: all path variables should be normalized (forward slashes) ASAP to conform with URI / URL standards. Therefore, this change normalizes rootPath.

Before this change, rootPath (Windows) uses backslashes which causes comparison on line 72 (in method mapFileToUrl) to fail. Attempts to register an account throws error (line 77). Note that the "path" variable for individual resources has already been explicitely normalized. (line 71)

Unlike "path", rootPath is constant and this correction takes place only once in the constructor. This has nearly zero impact on performance. Tested on Windows. This should have no impact on other operating systems such as Linux.

TEST performed on Windows: Ran node-solid-server on Windows. Registered new account. Logged in.

### CODE REVIEW

File: lib/resource-mapper.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ class ResourceMapper {
   async mapFileToUrl ({ path, hostname }) {
     // Remove the root path if specified
     path = path.replace(/\\/g, '/')
-    if (path.startsWith(this._rootPath)) {
+	if (path.startsWith(this._rootPath)) {
```
Comment (michielbdejong) [inline lib/resource-mapper.js:72]: please undo this indent, it was correct with the `i` of if under the `p` of path

**Comment by michielbdejong** [comment]
Failing in https://travis-ci.com/github/solid/node-solid-server/jobs/407056730#L287

File: lib/resource-mapper.js (hunk 1 above)
Comment (highlevellogic) [reply lib/resource-mapper.js]: Done.

Comment (bourgeoa) [reply lib/resource-mapper.js]: You added a ";". Lint impose a very strict style. There should be no ;

Comment (highlevellogic) [reply lib/resource-mapper.js]: OK, I didn't realize. I've removed the ;

**Comment by bourgeoa** [comment]
@michielbdejong 
Apparently all tests passes but there is `continuous integration expected`. what does it mean ?
I found this https://travis-ci.community/t/pr-build-status-stuck-at-continuous-integration-travis-ci-expected-waiting-for-status-to-be-reported/7220

**Comment by highlevellogic** [comment]
Continuous integration generally means frequent merges. It looks like this is just waiting for review.

**Comment by michielbdejong** [comment]
yes the build is green now

**Review by michielbdejong — COMMENTED** [review]
I see your explanation of why this change is necessary makes sense but it fixes a bug that I haven't been able to reproduce yet.
And this change looks quite scary because it's in the resource mapper.

**Comment by highlevellogic** [comment]
Officially, all testing is on Linux - which is presumably why you can't reproduce the problem. 
I'd like to suggest that the change isn't so scarey if it tests on Linux in the way described - i.e. has no effect.
No harm. No foul.
As for Windows - I think you can see what's happening just by looking at it. 
Two things are being compared: path and rootPath.
path was already being normalized with .replace(/\\/g, '/')
now rootPath is being normalize in exactly the same way.

The actual change isn't so scarey.

**Comment by highlevellogic** [comment]
Or you could do a simulation of the problem on Linux by transforming both path and rootPath to backward slashes and then see what happens when only one is changed to forward slashes.

**Comment by highlevellogic** [comment]
If you think it would help (to document or whatever) I can set up a separate simple demonstration for node.js ???

**Comment by highlevellogic** [comment]
Reproducing the problem

To emulate the problem on Linux, change line 33 from 
    this._rootPath = this._removeTrailingSlash(rootPath)
to 
    this._rootPath = this._removeTrailingSlash(rootPath).replace(/\//g,'\\')

Test
then add the fix
 this._rootPath = this._removeTrailingSlash(rootPath).replace(/\//g,'\\').replace(/\\/g, '/')

Actual new code (just the fix):
this._rootPath = this._removeTrailingSlash(rootPath).replace(/\\/g, '/')

**Comment by highlevellogic** [comment]
Would be nice to get this merged. Looks like more fixes may be coming in the same file.

**Comment by highlevellogic** [comment]
With one more fix to this file, it looks like the data browser is now working well on Windows.
I will submit a new PR with both fixes included.

**Pull request closed without merging**